File: src/main/java/teampearl/cvgenerator/back_end/presentation_layer/dto/TrainingDTO.java
package teampearl.cvgenerator.back_end.presentation_layer.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

public class TrainingDTO {
    @JsonProperty("id")
    private Integer id;

    @JsonProperty("name")
    private String name;

    @JsonProperty("instituteId")
    private InstituteDTO institute;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public InstituteDTO getInstitute() {
        return institute;
    }

    public void setInstitute(InstituteDTO institute) {
        this.institute = institute;
    }
}

File: src/app/dto/TrainingDTO.ts
import { DTO } from './DTO';
import { InstituteDTO } from './InstituteDTO';


export class TrainingDTO implements DTO{
  id!: number;
  name!: string;
  institute!: InstituteDTO;

  constructor() {
    this.institute = new InstituteDTO;
  }
}
